PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:3 + add:0, change matches:
-    _inherit = ['loyalty.program']
-    _inherit = ['loyalty.reward']
-    _inherit = ['sale.order']

Total Changes: 6

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/sale_loyalty_delivery/models/loyalty_program.py
+++ b/addons/sale_loyalty_delivery/models/loyalty_program.py
@@ -4,7 +4,7 @@ from odoo import _, api, models
 
 
 class LoyaltyProgram(models.Model):
-    _inherit = ['loyalty.program']
+    _inherit = 'loyalty.program'
 
     @api.model
     def _program_type_default_values(self):

--- a/addons/sale_loyalty_delivery/models/loyalty_reward.py
+++ b/addons/sale_loyalty_delivery/models/loyalty_reward.py
@@ -4,7 +4,7 @@ from odoo import _, fields, models
 
 
 class LoyaltyReward(models.Model):
-    _inherit = ['loyalty.reward']
+    _inherit = 'loyalty.reward'
 
     reward_type = fields.Selection(
         selection_add=[('shipping', 'Free Shipping')],

--- a/addons/sale_loyalty_delivery/models/sale_order.py
+++ b/addons/sale_loyalty_delivery/models/sale_order.py
@@ -5,7 +5,7 @@ from odoo.fields import Command
 
 
 class SaleOrder(models.Model):
-    _inherit = ['sale.order']
+    _inherit = 'sale.order'
 
     # delivery overrides
 
